site/content/contribute/webapp/build-component.md

```diff
@@ -23,89 +23,92 @@ The components for the webapp currently live in the `components/` directory in t
 
 These requirements are discussed in more detail in the following sections.
 
+## Designing the Component
 
-## Designing your Component
+The most important part of designing your component is deciding on what the props will be. Props are very much the API for your component. Think of them as a contract between your component and the users.
 
-The most important part of designing your component is deciding on what the props will be. Props are very much the API for your component, think of them as a contract between your component and the users.
-
-Props are read-only variables that get passed down to your component either directly from a parent component or from an `index.js` container that pulls some data from the Redux store. More on that later, for now you can see all the different prop types here: https://facebook.github.io/react/docs/typechecking-with-proptypes.html#proptypes.
+Props are read-only variables that get passed down to your component either directly from a parent component or from an `index.ts` container that connects the component to the Redux store.
 
 How do you decide what props your component should have? Think about what your component is trying to display to the user. Any data you need to accomplish that should be part of the props.
 
 As an example, let's imagine we're building an `ItemList` component with the purpose of displaying a list of items. The props for such a component might look like:
 
-```javascript
-static propTypes = {
+```typescript
+type Props = {
     /**
-    * Sets the title of the list
-    */
-    title: PropTypes.string,
-
+     * The title of the list
+     */
+    title?: string;
+    
     /**
-    * An array of item components to display
-    */
-    items: PropTypes.arrayOf(PropTypes.object).isRequired
+     * An array of items to display
+     */
+    items: ListItem[];
 }
 ```
 
-The `title` prop would control the string to display for the title, while `items` would be an array of the item objects we would like to display. Note that `items` has `.isRequired` appended while `title` does not. This means that to render our component it must have the `items` prop set, while it can still render if we don't set `title`. Use this to provide optional props.
+The `title` prop would control the string to display for the title, while `items` would be an array of the item objects we would like to display. Note that `items` is required while `title` is optional.
```

Comment:
not exactly sure if this is correct (from my end), but it sounds as if it should be without the `for`

```suggestion
The `title` prop would control the string to display the title, while `items` would be an array of the item objects we would like to display. Note that `items` is required while `title` is optional.
```

Reply:
You're right that it sounds a bit weird with the for, but I think it's similarly weird without it. I think that's caused by the "would"s in that sentence. How does this sound instead?
```suggestion
The `title` prop is a string that's displayed as the title of the list, and `items` is an array of objects that make up the contents of the list. Note that `items` is required while `title` is optional.
```